### ISSUE TITLE
error in config no longer produces error in stats.compilation.errors

### ISSUE THREAD

**Issue by adrianhall** [stylelint/stylelint#876]
> Describe the issue. Is it a bug or a feature request (new rule, new option, etc.)?

REGRESSION

In versions v4.4.x and prior, if there was an error in the configuration, the stylelint API would add an error to stats.compilation.errors with the information about the configuration.  In version v4.5.0, this is no longer the case.

> Which rule, if any, is this issue related to?

No specific rule

> What CSS is needed to reproduce this issue?

```css
body {
    font-size: 8px;
}
```

(Actually, any CSS can be used)

> What stylelint configuration is needed to reproduce this issue?

```json
module.exports = {
    rules: {
        // This next line is an error in the rule
        'color-hex-case': 'error',
        'selector-list-comma-newline-after': [ 'always', { warn: true } ],
        'unit-blacklist': [ 'em' ],
        'unit-whitelist': [ 'px' ]
    }
}
```

> Which version of stylelint are you using?

`4.5.0`

> How are you running stylelint: CLI, PostCSS plugin, Node API?

Webpack with stylelint-loader (I am the maintainer).

Test for this condition is in the stylelint-loader mocha tests
stylelint-loader repository: https://github.com/adrianhall/stylelint-loader

> Does your issue relate to non-standard syntax (e.g. SCSS, nesting, etc.)?

No

> What did you expect to happen?

Expected error to be added to stats.compilation.errors

> What actually happened (e.g. what warnings or errors you are getting)?

No error was produced.

**Comment by davidtheclark** [comment]
What is `stats.compilation.errors`? I don't know what it is, so never made
the code put anything there on purpose :)

**Comment by adrianhall** [comment]
It's where all the errors and warnings used to go :-)

There is a stats.compilation.errors & stats.compilation.warnings - the old API (when I first wrote stylelint-loader) placed all the errors there.  Most of them still end up there (particularly those that are the result of a CSS/SCSS error).  

Take a look at the stylelint-loader code - the actual loader hasn't changed for a long time and it isn't that complex a code since it's just a wrapper around the stylelint SDK - only the peerDependencies change with each release of stylelint.

**Comment by davidtheclark** [comment]
Sorry, it's still not clear to me. stylelint never put errors or warnings in any `stats.compilation` object. This must be a webpack construct you're referring to, and something about your loader was moving things into that object. So with that in mind: what could have changed on our end to make this difference? Later I can double-check to see if there was some change to the way that configuration errors get reported (I don't think there was, though). But I can't really know how to fix this or what needs fixing unless you can give me more details about how the loader works and what is needed that is not being provided.

**Comment by adrianhall** [comment]
Sorry - yes, there is a webpack construct here  When stylelint.lint() returns a Promise, there is an array of results - each result had a warnings & errors array - these were moved into the webpack warnings & errors.

What changed on your end was that the config file erorrs and warnings got moved into the deprecations & invalidOptionWarnings properties of the result object.  I've adjusted my webpack loader accordingly.

Closing this now I've figured it out.  However, would be nice to add a line/column to the deprecations & invalidOptionWarnings objects so that we know where the actual error is.

**Comment by davidtheclark** [comment]
@adrianhall Do you mean line/column of the configuration file?

**Comment by adrianhall** [comment]
@davidtheclark Yes - when you have a couple hundred lines of configuration, it would be helpful to understand where the error is.

**Comment by jeddy3** [comment]
Shall we open a new issue about that?

**Comment by davidtheclark** [comment]
No, I don't think so. Hate to just disagree, but I think that since we have informative error messages and editors have CTRL+F, it is not difficult at all to find the problem no matter how large the configuration file. (If there is an error message that is so unclear you can't easily find the offending line, that's a problem with the error message.) Besides, I do not think it is feasible.

**Comment by jeddy3** [comment]
> Besides, I do not think it is feasible.

I was wondering about that. 

Leaving things as they are sgtm.
